server/dummyControllers/requestController.js

```diff
@@ -0,0 +1,39 @@
+import requests from './../dummyModels/request';
+import users from './../dummyModels/user';
+
+/**
+ * it is a class that control all request api;
+ */
+class RequestController {
+  /**
+     * it GET all requests
+     * @param {string} req
+     * @param {string} res
+     * @returns {object} object
+     */
+  static getAllRequests(req, res) {
+    const admin = users.filter(user => user.role === 'admin');
+    // check if user is admin to get all request
+    if (!admin) {
```

Comment:
Is there a reason why an admin is denied rights to view a user requests?

Reply:
The admin has the only right to get all requests of the users

Reply:
That's not foresight enough, let's say the app grew and admin can filter requests based on a user from the admin dashboard.  You have either edit this function**(violates the open and close principle of SOLID)** or create another function which will just be repeating code**(violates Single responsibility principle)**.

Reply:
An addendum is the admin has the ultimate rights to do anything on the system.